src/Storage/Token.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Storage;
+
+use DateTime;
+use DateTimeInterface;
+use Lcobucci\JWT\Token as TokenInterface;
+
+/**
+ * Basic structure of the JWT
+ *
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ * @since 0.1.0
+ */
+final class Token implements TokenInterface
+{
+    /**
+     * The token headers
+     *
+     * @var DataSet
+     */
+    private $headers;
+
+    /**
+     * The token claim set
+     *
+     * @var DataSet
+     */
+    private $claims;
+
+    /**
+     * The token signature
+     *
+     * @var Signature|null
+     */
+    private $signature;
+
+    /**
+     * Initializes the object
+     *
+     * @param DataSet $headers
+     * @param DataSet $claims
+     * @param Signature|null $signature
+     */
+    public function __construct(
```

Comment:
Not caring about BC breaks here.
The object is only created by the `Builder`, people shouldn't instantiate it.
We can add named constructors here... `Token::unsigned()` and `Token::signed()`?